src/components/input/Select.tsx

```diff
@@ -327,16 +324,12 @@ function SelectMain<T>({
   containerClasses,
   onListboxScroll,
   right = false,
-  multiple = false,
+  multiple,
   'aria-label': ariaLabel,
   'aria-labelledby': ariaLabelledBy,
   /* eslint-disable-next-line no-prototype-builtins */
   listboxAsPopover = HTMLElement.prototype.hasOwnProperty('popover'),
-}: SelectNextProps<T>) {
-  if (multiple && !Array.isArray(value)) {
-    throw new Error('When `multiple` is true, the value must be an array');
-  }
```

Comment:
There's no need to check this anymore, as we enforce it via static types.

* `Select` allows single-item values and hardcodes `multiple: false`.
* `MultiSelect` allows array values and hardcodes `multiple: true`.